src/electionguard/ballot_compact.py

```diff
@@ -0,0 +1,186 @@
+from dataclasses import dataclass
+from typing import Dict, List
+
+from electionguard.group import ElementModQ, int_to_q_unchecked
+
+from .ballot import (
+    SubmittedBallot,
+    ExtendedData,
+    PlaintextBallot,
+    PlaintextBallotContest,
+    PlaintextBallotSelection,
+    create_ballot_hash,
+)
+from .ballot_box import BallotBoxState
+from .election import (
+    CiphertextElectionContext,
+    ContestDescriptionWithPlaceholders,
+    InternalElectionDescription,
+)
+from .encrypt import encrypt_ballot_contests
+from .utils import get_optional
+
+YES_VOTE = 1
+NO_VOTE = 0
+
+
+@dataclass
+class CompactPlaintextBallot:
+    """A compact plaintext representation of ballot minimized for data size"""
+
+    object_id: str
+    style_id: str
+    selections: List[bool]
+    extended_data: Dict[int, ExtendedData]
+
+
+@dataclass
+class CompactSubmittedBallot:
+    """A compact submitted ballot minimized for data size"""
+
+    compact_plaintext_ballot: CompactPlaintextBallot
+    timestamp: int
+    ballot_nonce: int
```

Comment:
I'm holding onto the actual `ElementModQ` references in the c++ objects because the serialization needs to export in hex string and its already supported in those objects.